packages/@aws-cdk/cdk/lib/construct.ts

```diff
@@ -83,7 +82,7 @@ export class ConstructNode {
     // escape any path separators so they don't wreck havoc
     this.id = this._escapePathSeparator(this.id);
 
-    if (unresolved(id)) {
+    if (Token.unresolved(id)) {
```

Comment:
isToken

Reply:
Why don't we just remove `Token.unresolved()` to turn this into a compilation error?